Anicorn doesn't understand stale PID file

Hi 👻

I had a small problem with PID file. Maybe you'll want to add anti-measures.

Server was turned off. Unicorn left behind a PID file. It was invalid. But Anicorn kept trying to work with it:

```
[ 2018/02/10 20:14:03 ] [WARN] Looks like Unicorn is down. Restarting...
[ 2018/02/10 20:14:03 ] Unicorn already works, restarting...
[ 2018/02/10 20:14:03 ] Restarting Unicorn (PID: 3577)...
/usr/bin/anicorn: line 210: kill: (3577) - No such process
[ 2018/02/10 20:14:19 ] [ERROR] Can't restart Unicorn
[ 2018/02/10 20:14:24 ] [WARN] Looks like Unicorn is down. Restarting...
[ 2018/02/10 20:14:24 ] Unicorn already works, restarting...
[ 2018/02/10 20:14:24 ] Restarting Unicorn (PID: 3577)...
/usr/bin/anicorn: line 210: kill: (3577) - No such process
[ 2018/02/10 20:14:40 ] [ERROR] Can't restart Unicorn
[ 2018/02/10 20:14:45 ] [WARN] Looks like Unicorn is down. Restarting...
[ 2018/02/10 20:14:45 ] Unicorn already works, restarting...
[ 2018/02/10 20:14:45 ] Restarting Unicorn (PID: 3577)...
/usr/bin/anicorn: line 210: kill: (3577) - No such process
[ 2018/02/10 20:15:01 ] [ERROR] Can't restart Unicorn
```

I was able to break this cycle only by manually removing the PID file. After that Anicorn did everything correctly:

```
[ 2018/02/10 20:15:48 ] [WARN] Looks like Unicorn is down. Restarting...
[ 2018/02/10 20:15:48 ] Starting Unicorn...
[ 2018/02/10 20:16:04 ] Unicorn successfully started (PID: 31271)
```

Hope this helps!

--- a/SOURCES/anicorn
+++ b/SOURCES/anicorn
@@ -376,7 +376,7 @@ getPID() {
     return
   fi
 
-  echo $pid
+  echo "$pid"
 }
 
 # Extract path to PID file from Unicorn config

--- a/SOURCES/anicorn
+++ b/SOURCES/anicorn
@@ -357,17 +357,26 @@ monitorStoppedUnicorn() {
   kill -TERM "$pid"
 }
 
-# Read PID from PID file
+# Read PID from PID file and validate it
 #
 # Code: No
 # Echo: PID (Number)
 getPID() {
+  local pid
+
   if [[ ! -e "$pid_file" ]] ; then
     echo ""
     return
   fi
 
-  tr -d '\n\r' < "$pid_file"
+  pid=$(tr -d '\n\r' < "$pid_file")
+
+  if [[ ! -e "/proc/$pid" ]] ; then
+    echo ""
+    return
+  fi
+
+  echo $pid
 }
 
 # Extract path to PID file from Unicorn config

